util.c

```diff
@@ -722,9 +725,9 @@ int insert_image(struct image *image, struct image *sub,
 					image_error(image, "short write (%d vs %d)\n", w, r);
 				goto out;
 			}
-			size -= now;
-			offset += now;
-			in_pos += now;
+			size -= w;
+			offset += w;
+			in_pos += w;
```

Comment:
ack to this part, obviously. I'll have to read the remaining commits to comment on the other hunk.